src/Tasks/ResolveAssemblyReferences/Server/RarController.cs

```diff
@@ -130,9 +126,32 @@ public async Task<int> StartAsync(CancellationToken cancellationToken = default)
             return 0;
         }
 
-        private async Task HandleClientAsync(Stream serverStream, CancellationToken cancellationToken = default)
+        private async Task<Stream?> ConnectAsync(CancellationToken cancellationToken = default)
+        {
+            Stream serverStream = GetStream(_pipeName);
+
+            if (!(serverStream is NamedPipeServerStream pipeServerStream))
+            {
+                return serverStream;
+            }
+
+            await pipeServerStream.WaitForConnectionAsync(cancellationToken).ConfigureAwait(false);
+
+            if (_validateHandshakeCallback(pipeServerStream, ValidationTimeout))
+            {
+                return pipeServerStream;
+            }
+            Console.WriteLine("ERROR: Didn't validate handshake");
```

Comment:
?

Reply:
Forgot to delete this, old fashioned printf debugging :D